syntax = "proto2";

enum Protocol {
    REGISTER         = 130000; //注册
    LOGIN            = 130001; //登录
    REQUEST_AUTHCODE = 130002; //请求验证码
    VERIFY_AUTHCODE  = 130003; //验证手机号
    FRESH_LOCATION   = 130004; //更新位置
}

enum Error_Code {
    SUCCESS = 0;
    ERROR_PASSWORD_INVALID   = 130000; //密码错误
    ERROR_PASSWORD_NOT_EQUAL = 130001; //两个密码不一致
    ERROR_WASHER_EXIST       = 130002; //商家已存在
    ERROR_AUTHCODE_INVALID   = 130003; //验证码不对
    ERROR_AUTHCODE_EXPIRED   = 130004; //验证码过期
    ERROR_PHONE_INVALID      = 130005; //无效手机号
    ERROR_WASHER_NOT_FOUND   = 130006; //找不到商家
    ERROR_SIGNATURE_EXPIRED  = 130007; //签名过期
    ERROR_SIGNATURE_INVALID  = 130008; //签名错误
    ERROR_KICKOUT            = 130009; //踢下线 
    ERROR_BADREQEUST         = 130010; //非法请求
}

message Washer {
    required string id           = 1;
    optional string nick         = 2;
    optional string phone        = 3;
    optional string avatar       = 4;
    optional float  level        = 5;
    optional int32  reg_time     = 6;
    optional int32  last_login   = 7;
    optional int32  status       = 8;
    optional string secret       = 9;
}

message Request_Authcode_Request {
    required string phone = 1;
    required string signature = 2;
}

message Request_Authcode_Response {
    optional int32 authcode = 1;
    required Error_Code error_code = 2;
}

message Register_Request {
    required string phone     = 1;
    required int32  authcode  = 2;
    required string password  = 3;
    required string password2 = 4;
    required string nick      = 5;
    required string signature = 6;
    optional string avatar    = 7;
}

message Register_Response {
    optional Washer washer         = 1;
    required Error_Code error_code = 2;
}

message Login_Request {
    required string phone     = 1;
    optional string password  = 2;
    optional string uuid      = 3;
    optional string signature = 4;
}

message Login_Response {
    optional Washer washer         = 1;
    required Error_Code error_code = 3;
}

message Verify_Authcode_Request {
    required string phone     = 1;
    required int32  authcode  = 2;
    required string signature = 3;
}

message Verify_Authcode_Response {
    required Error_Code error_code = 1;
}

message Fresh_Location_Request {
    required int32  city_code = 1; 
    required float  longitude = 2;
    required float  latitude  = 3; 
}

message Fresh_Location_Response {
    required Error_Code error_code = 1;
}

